src/PlayerAnalyticsConnector.ts

```diff
@@ -56,51 +60,43 @@ export class PlayerAnalyticsConnector {
 
   private initiateVideoEventFilter() {
     if (!this.player) return;
-    this.videoEventFilter = new VideoEventFilter(this.player);
-    this.videoEventFilter.addEventListener(
-      "*",
-      (this.videoEventListener = (event: PlayerEvents) => {
-        if (!Object.keys(EPASEvents).includes(event)) return;
+    this.videoEventFilter = getMediaEventFilter({
+      mediaElement: this.player,
+      mp4Mode: false,
+      callback: (event: FilteredMediaEvent) => {
         let eventType;
         const extraData = {};
-        if (!event) return;
         switch (event) {
-          case PlayerEvents.Loaded:
+          case FilteredMediaEvent.LOADED:
             eventType = EPASEvents.loaded;
             break;
-          case PlayerEvents.Loading:
-            eventType = EPASEvents.loading;
-            break;
-          case PlayerEvents.Play:
-          case PlayerEvents.Resume:
+          case FilteredMediaEvent.PLAY:
             eventType = EPASEvents.play;
             this.startInterval();
             break;
-          case PlayerEvents.Pause:
+          case FilteredMediaEvent.PLAYING:
+            eventType = EPASEvents.resume;
+            break;
```

Comment:
just for clarity, sinec some of this does not look quite correct

PLAY means a request to play has been made by the user (a call to videoElement.play()), but video is not yet playing

PLAYING means video has started playing (for the first time, or after a PLAY request was made)

EPAS makes no difference between PLAYING and RESUME (where resume would usually mean any playing events after the first one). The old video-event-filter made this distinction, and it seems that has leaked into the EPASEvents implementation even if it's not part of the spec.

For example, this would be the event sequence if video started buffering, and the user pressed pause/play in frustration:

BUFFERING - PAUSE - PLAY (video cant start yet, no playing event) - BUFFERED - PLAYING